build.ps1

```diff
@@ -1,5 +1,239 @@
+        $newContent = $content -replace $netCoreAppOldVersion, $netCoreAppNewVersion;
+        $newContent2 = $newContent -replace $oldVersion, $newVersion;
+        Set-Content $fileName $newContent2;
+    }
+}
+
+if ($clean)
+{
+    Write-Host ""
+    Write-Host "============================"
+    Write-Host "Clean"
+    Write-Host ""
+
+    $projects = $buildConfiguration.SelectNodes("root/projects/src/project");
+    foreach($project in $projects) {
+        $name = $project.name;
+        if (Test-Path("$root\src\$name\bin"))
+        {
+            Write-Host ">>> Remove-Item -Recurse -Force $root\src\$name\bin"
+            Remove-Item  -Recurse -Force $root\src\$name\bin
+        }
+
+        if (Test-Path("$root\src\$name\obj"))
+        {
+            Write-Host ">>> Remove-Item -Recurse -Force $root\src\$name\obj"
+            Remove-Item  -Recurse -Force $root\src\$name\obj
+        }
+    }
+
+    $testProjects = $buildConfiguration.SelectNodes("root/projects/test/project")
+    foreach ($testProject in $testProjects) {
+        $name = $testProject.name;
+        if (Test-Path("$root\test\$name\bin"))
+        {
+            Write-Host ">>> Remove-Item -Recurse -Force $root\test\$name\bin"
+            Remove-Item  -Recurse -Force $root\test\$name\bin
+        }
+
+        if (Test-Path("$root\test\$name\obj"))
+        {
+            Write-Host ">>> Remove-Item -Recurse -Force $root\test\$name\obj"
+            Remove-Item  -Recurse -Force $root\test\$name\obj
+        }
+    }
+
+    if (Test-Path $artifactsRoot)
+    {
+        Write-Host ">>> Remove-Item -Recurse -Force $artifactsRoot"
+        Remove-Item  -Recurse -Force $artifactsRoot
+    }
+
+    if (Test-Path $dotnetInstallDir)
+    {
+        Write-Host ">>> Remove-Item -Recurse -Force $dotnetInstallDir"
+        Remove-Item  -Recurse -Force $dotnetInstallDir
+    }
+}
+
+if (!(Test-Path $artifactsRoot))
+{
+    Write-Host ">>> mkdir $artifactsRoot | Out-Null"
+    mkdir $artifactsRoot | Out-Null
+}
+
+if ($installDotnet -eq "YES")
+{
+    Write-Host ""
+    Write-Host "============================"
+    Write-Host "Install dotnet"
+    Write-Host "dotnetVersion = $dotnetVersion"
+    Write-Host "dotnetLocalInstallFolder = $dotnetInstallDir"
+    Write-Host "dotnetexe = $dotnetexe"
+    Write-Host ""
+
+    if (!(Test-Path $dotnetInstallDir))
+    {
+        Write-Host "mkdir $dotnetInstallDir | Out-Null"
+        mkdir $dotnetInstallDir | Out-Null
+    }
+
+    # download script to install dotnet
+    Write-Host "Invoke-WebRequest $dotnetUrl -OutFile $dotnetInstallDir\dotnet-install.ps1"
+    Invoke-WebRequest $dotnetUrl -OutFile "$dotnetInstallDir\dotnet-install.ps1"
+
+    & "$dotnetInstallDir\dotnet-install.ps1" -Channel $dotnetChannel -Version $dotnetVersion -Architecture x64 -InstallDir $dotnetInstallDir -Verbose
+    if($LASTEXITCODE -ne 0)
+    {
+        throw "Failed to install dotnet"
+    }
+}
+
+if ($updateCoreFxVersion -eq "YES")
```

Comment:
We can change the mapping in a future commit.
User the flag -updateCoreFxVersion NO